fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -638,8 +638,14 @@ public PlanFragment visitPhysicalOlapScan(OptExpression optExpr, ExecPlan contex
                             .getBackendIdByHost(FrontendOptions.getLocalHostAddress());
                 }
 
-                for (Long partitionId : node.getSelectedPartitionId()) {
+                List<Long> selectedNonEmptyPartitionIds = node.getSelectedPartitionId().stream().filter(p -> {
+                    List<Long> selectTabletIds = scanNode.getPartitionToScanTabletMap().get(p);
+                    return selectTabletIds != null && !selectTabletIds.isEmpty();
+                }).collect(Collectors.toList());
+                scanNode.setSelectedPartitionIds(selectedNonEmptyPartitionIds);
+                for (Long partitionId : scanNode.getSelectedPartitionIds()) {
                     List<Long> selectTabletIds = scanNode.getPartitionToScanTabletMap().get(partitionId);
+                    Preconditions.checkState(selectTabletIds != null && !selectTabletIds.isEmpty());
```

Comment:
If all the user-provided tablet_ids are invalid, will here throw an `Unknown Error` to the MySQL client?